src/actions/bookingDaysOffActions.js

```diff
@@ -60,119 +43,158 @@ const validateBody = (body) => {
         startDate = new Date(startDate);
         endDate = new Date(endDate);
-
+        
         let daysOffArray = [];
-
+        
         if (startDate > endDate) {
             reject({
-                "status": 400,
-                "message": "End date lower than start date"
+                status: 400,
+                message: "End date lower than start date"
             });
         }
-
+        
         endDate.setDate(endDate.getDate() + 1);
         for (; startDate < endDate; startDate.setDate(startDate.getDate() + 1)) {
             if (startDate.getDay() > 0 && startDate.getDay() < 6) {
-                daysOffArray.push(`${startDate.getFullYear()}-${formatDate(startDate.getUTCMonth() + 1)}-${formatDate(startDate.getDate())}`);
+                daysOffArray.push(
+                    `${startDate.getFullYear()}-${formatDate(
+                        startDate.getUTCMonth() + 1
+                    )}-${formatDate(startDate.getDate())}`
+                );
             }
         }
-
+        
         resolve(daysOffArray);
-    })
+    });
+};
+
+const createUpdatedArray = (alreadyBookedDays, arrayOfDaysOff) => {
+    let updatedDaysOff = [];
+    for(let i=0; i<arrayOfDaysOff.length; i++){
+        if(alreadyBookedDays.indexOf(arrayOfDaysOff[i]) === -1) 
+            updatedDaysOff.push(arrayOfDaysOff[i])
+    }
+    return updatedDaysOff
 }
 
-const createNewDaysOffJSON = (daysOff, userId, daysOffArray) => {
-    return new Promise((resolve, reject) => {
-        resolve({
-            id: newId(daysOff),
-            userId: userId,
-            daysOff: daysOffArray
-        });
+const removeDuplicateDaysOff = (userId,arrayOfDaysOff) => {
+   return new Promise((resolve,reject) => {
+        getAlreadyBookedDays(userId)
+            .then(alreadyBookedDays => {
+                updatedDaysOff = createUpdatedArray(alreadyBookedDays, arrayOfDaysOff)
+                if(updatedDaysOff.length === 0){
+                    reject({
+                        status: 422,
+                        message: "This days are already booked"
+                    })
+                }
+                resolve(updatedDaysOff);
+            })
+            .catch(err => {
+                resolve(arrayOfDaysOff)
+            })
 
-    })
+        })
+    
 }
 
-const addNewDaysOffJSONInArray = (daysOff, newDaysOffJSON) => {
+const createAndSaveDaysOffFlow = (userId, daysOffArray) => {
+    return createNewDaysOffJSON(userId, daysOffArray).then(saveDayOffInDb);
+};
+
+const saveDayOffInDb = newDaysOff => {
     return new Promise((resolve, reject) => {
-        daysOff.push(newDaysOffJSON);
-        resolve(JSON.stringify(daysOff));
-    })
-}
+        let newDays = new DaysOff(newDaysOff);
+        newDays.save((err, data) => {
+            if (err) reject(err);
+            resolve(data);
+        });
+    });
+};
 
-const sendResponse = (response, newDaysOffJSON) => {
+const createNewDaysOffJSON = (userId, daysOffArray) => {
```

Comment:
This function is not creating a JSON, but a javascript object wrapped in a promise. It should either be renamed, or (since it doesn't do that much) remove it completely